app/src/main/java/com/example/sharingang/items/FirestoreItemRepository.kt

```diff
@@ -0,0 +1,100 @@
+package com.example.sharingang.items
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.example.sharingang.BuildConfig
+import com.google.firebase.firestore.Query
+import com.google.firebase.firestore.ktx.firestore
+import com.google.firebase.firestore.ktx.firestoreSettings
+import com.google.firebase.ktx.Firebase
+import kotlinx.coroutines.tasks.await
+import java.lang.Exception
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val TAG = "FirestoreItemRepository"
+
+@Singleton
+/**
+ * Implementation of ItemRepository using the Firestore database
+ *
+ * During development it requires running the Firebase emulator (see README.md)
+ */
+class FirestoreItemRepository @Inject constructor() :
+    ItemRepository {
+    private val firestore = Firebase.firestore
+    private val collectionName = "items"
+
+    init {
+        if (BuildConfig.DEBUG) {
+            Log.d(TAG, "Using Firebase emulator.")
+            // 10.0.2.2 is the special IP address to connect to the 'localhost' of
+            // the host computer from an Android emulator.
+            firestore.useEmulator("10.0.2.2", 8080)
+
+            // Because the Firebase emulator doesn't persist data, we disable the local persistence
+            // to avoid conflicting data.
+            firestore.firestoreSettings = firestoreSettings {
+                isPersistenceEnabled = false
+            }
+        } else {
+            Log.d(TAG, "Using production Firebase.")
+        }
+    }
+
+    override suspend fun getItem(id: String): Item? {
+        val document = firestore.collection(collectionName)
+            .document(id)
+            .get()
+            .await()
+
+        return if (document != null) {
+            document.toObject(Item::class.java)
+        } else {
+            Log.d(TAG, "No Item with ID = $id")
+            null
+        }
```

Comment:
IMO, separate logging from return value here. Plus, you can use Kotlin's neat nullable features :)

```suggestion
        if (document != null) {
          Log.d(TAG, "No Item with ID = $id")
        }
        return document?.toObject(Item::class.java)
```

Reply:
Thanks, it's much more readable :)